### DISCUSSION TITLE
How to create a custom filter with non-std argument?

### DISCUSSION THREAD

**Discussion by hongquan** [mitsuhiko/minijinja#312] [Q&A] [answered]
For example, I want to implement a filter:

```rust
fn get_detail_url(created_at: OffsetDateTime, slug: String) -> String;
```
where `OffsetDateTime` is from [time](https://crates.io/crates/time) crate.

**Comment by mitsuhiko** [comment, marked as answer]
`Value` holds a serde compatible representation. Depending on how the value serializes you can extract out of it. You can then use a manual conversion to get to the object you want. For `time` in particular the `minijinja-contrib` module has some examples: https://github.com/mitsuhiko/minijinja/blob/32894ab55d2a23ef8e98117c962dab88e83e6eed/minijinja-contrib/src/filters/datetime.rs#L15

```rust
pub fn datetimeformat(value: Value) -> Result<String, Error> {
    let datetime = value_to_datetime(value)?;
    // ...
}
```

**Comment by hongquan** [comment]
Thanks.
